src/inmanta/env.py

```diff
@@ -1183,13 +1184,17 @@ def is_owned_by(self, owners: abc.Set[str]) -> bool:
                 (
                     []
                     if strict_scope is None
-                    else (dist_info.key for dist_info in pkg_resources.working_set if strict_scope.fullmatch(dist_info.key))
+                    else (
+                        packaging.utils.canonicalize_name(dist_info.name)
+                        for dist_info in distributions()
+                        if strict_scope.fullmatch(dist_info.name)
+                    )
                 ),
-                (requirement.requirement.name for requirement in inmanta_constraints),
-                (requirement.requirement.name for requirement in extra_constraints),
+                (cast(NormalizedName, requirement.requirement.name) for requirement in inmanta_constraints),
```

Comment:
This cast is required because the typing doesn't know that a `CanonicalRequirement` always has a `NormalizedName` ...

There may be a away around this, but I don't know it. 

My best guess would be to do the following

```python
if typing.TYPE_CHECKING:
    class CanonicalRequirement(packaging.requirements.Requirement):
        name: NormalizedName

else:
    CanonicalRequirement = typing.NewType("CanonicalRequirement", packaging.requirements.Requirement)
    """
    A CanonicalRequirement is a packaging.requirements.Requirement except that the name of this Requirement is canonicalized, which
    allows us to compare names without dealing afterwards with the format of these requirements.
    """

```

But I don't know if that is acceptable?

Reply:
I did it anyways